app/controllers/v1/articles_controller.rb

```diff
@@ -12,19 +12,22 @@ def index
   end
 
   def show
-    if Article.exists?(id: params[:id])
-      article = Article.find(params[:id])
-      render json: article, serializer: Articles::IndexSerializer
+    if (current_user.role == 'subscriber') || (current_user.role == 'journalist')
```

Comment:
Are you using `enum` for the role?

if so you can shorten this by calling `current_user.subscriber` and `current_user.journalist`

if not you should :)

Reply:
We are currently not using the `enum` method but how long would you estimate it would take to change? Since this method seems to work and we are thus wondering if it is a priority to work on changing that rather than moving on to a new Feature.

Reply:
It would not take long.

You need to change your role from `string` to `integer` and add this line to your `user.rb`:

```
 enum role: { user: 0, subscriber: 1, journalist: 2 }
```

You may need to update some tests and seed file but that should be it